vcf-dumper/vcf-dumper-lib/src/test/java/uk/ac/ebi/eva/vcfdump/VariantToVariantContextConverterTest.java

```diff
@@ -289,35 +308,37 @@ public void severalNucleotidesDeletionInPosition1() {
         String variantLine = String
                 .join("\t", CHR_1, "1", "id", "ATTG", "G", "100", "PASS", ".", "GT", "0|0", "0|0", "0|1", "1|1", "1|1",
                       "0|1");
-        List<Variant> variants = variantFactory.create(variantSource, variantLine);
+        List<Variant> variants = variantFactory.create(CHR_1, STUDY_1 , variantLine);
         assertEquals(1, variants.size());
+        VariantWithSamplesAndAnnotation variantSA = new VariantWithSamplesAndAnnotation(variants.get(0), s1s6SampleList);
 
         // export variant
-        BiodataVariantToVariantContextConverter variantConverter =
-                new BiodataVariantToVariantContextConverter(Collections.singletonList(variantSource),
+        VariantToVariantContextConverter variantConverter =
+                new VariantToVariantContextConverter(Collections.singletonList(variantSource),
                                                             noSampleNamesConflictSampleNameCorrections);
-        VariantContext variantContext = variantConverter.transform(variants.get(0));
+        VariantContext variantContext = variantConverter.transform(variantSA);
         System.out.println(variantContext);
-        checkVariantContext(variantContext, CHR_1, 1, 4, "ATTG", "G", variants.get(0).getSourceEntries(), false);
+        checkVariantContext(variantContext, CHR_1, 1, 4, "ATTG", "G", variantSA.getSourceEntries(), false);
     }
 
     @Test
     public void twoStudiesNoConflictingNamesSingleVariant() {
         // create test variant, with two studies and samples with not conflicting names
-        Variant variant = new Variant(CHR_1, 1000, 1000, "T", "G");
+        VariantWithSamplesAndAnnotation variant = new VariantWithSamplesAndAnnotation(CHR_1, 1000, 1000, "T", "G");
 
-        VariantSource source1 = new VariantSource("testFile1", "file_1", "study_1", "testStudy1");
-        source1.setSamples(Arrays.asList("SX_1", "SX_2", "SX_3", "SX_4"));
-        VariantSourceEntry study1Entry = new VariantSourceEntry("file_1", "study_1");
+        VariantSource source1 = new VariantSource("testFile1", "file_1", "study_1", "testStudy1", null, null, null, null, null, null);
+        //source1.setSamples(Arrays.asList("SX_1", "SX_2", "SX_3", "SX_4"));
+        IVariantSourceEntry entry1 = new VariantSourceEntry("file_1", "study_1", null, "format");
+        VariantSourceEntryWithSampleNames study1Entry = new VariantSourceEntryWithSampleNames(entry1, s1s6SampleList);
```

Comment:
The sample names passed to this study entry are s1, s2, s3....but in the next block we are trying to add SX_1, SX2, SX_3... . Those "addGenotype" calls are actually doing nothing because there is a commented line with a "todo".  Also, some assertions that check the sample names in this test are commented, and that explains that the test pass.

We have to uncomment all those lines, to make the test fail, and then fix the tests (passing the correct sample name list). If those incomplete tests are masking some incorrect behaviour in the code, then the code should be fixed too.

Reply:
There are no commented-out assertions in this test class. Instead of adding samples via adder method, they are added using constructor.

Reply:
Variant source entry is receiving "s1s6SampleList" as a list of sample names, and then we are adding some genotypes using different sample names (SX_1, SX_2... instead of s1, s2...). The variant source objects also have no sample list (the parameter samplesPosition of variantSource is null).

The result of that is that when we transform the variant to a VariantContext object, that object has no genotypes. But there are genotypes, so we have to fix the test data to make the transformed variants have genotypes again. The checkVariantContext method will assert that they are correct.